Sources/TuistKit/ProjectGenerator/ProjectGenerator.swift

```diff
@@ -0,0 +1,124 @@
+import Basic
+import Foundation
+import TuistCore
+import TuistGenerator
+import TuistLoader
+
+protocol ProjectGenerating {
+    @discardableResult
+    func generate(path: AbsolutePath, projectOnly: Bool) throws -> AbsolutePath
+}
+
+protocol DetailedProjectGenerating {
```

Comment:
What about merging `DetailedProjectGenerating` and `ProjectGenerating` into a single protocol with a method that returns both, the `AbsolutePath` and the `Graph`? When I saw the class using an instanced of `DetailedProjectGenerating` I was confused because I didn't know what `Detail...` meant.

Reply:
The intention was to offer two interfaces to the same component a simple one and a more advanced one (the one that returns the graph). Perhaps it's it's causing more confusion than good here, will merge them as suggested 👍